### PR TITLE
Add context submit support for job-name, job-group, etc

### LINKED ISSUE #60: Distributed submit: enable specifying job name and job group

### PR DESCRIPTION
**Pull request by dlaboss**
* Add JobProperties defining new properties
* Enhance Vwap sample to enable specifying context config properties
* Enhance distributed submit to support new job context properties and
existing ContextProperties.TRACING
* Enhance all invokers {build,standalone,distributed} to trace their
full cmdline

Fixes #60

### CODE REVIEW

File: samples/java/functional/src/vwap/Vwap.java [hunk 1]
```diff
@@ -32,12 +37,15 @@
      * @param args
      */
     public static void main(String[] args) throws Exception {
+        String mode = args[0];
```
Comment (ddebrunner) [inline samples/java/functional/src/vwap/Vwap.java]: The context type is a 'type', not a 'mode'

File: samples/java/functional/src/simple/Util.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                            value = toTracingLevel(valStr);
+                        else if (item.equals(ContextProperties.KEEP_ARTIFACTS))
+                            value = Boolean.valueOf(valStr);
+                        else if (item.equals(JobProperties.OVERRIDE_RESOURCE_LOAD_PROTECTION))
+                            value = Boolean.valueOf(valStr);
+                        else if (item.equals(JobProperties.PRELOAD_APPLICATION_BUNDLES))
+                            value = Boolean.valueOf(valStr);
+                        
+                        config.put(item, value);
+                        System.out.println("Setting config: item="+item+" value="+value+"  [arg="+arg+"]");
```
Comment (ddebrunner) [inline samples/java/functional/src/simple/Util.java]: This should be through a logger, not System.out. System.out is for the application output.

**Comment by ddebrunner** [comment]
Do we need tests for this? E.g. verifying the data directory was set.

Code without tests will drop the code coverage numbers

**Comment by dlaboss** [comment]
Yeah, test would be good but wasn't quite sure how to tackle checking all/some of these.
I suppose a test could capture logging output and observe the correct things were set in the streamtool submitjob request.  I'll look into it.

**Comment by ddebrunner** [comment]
Added issue #65 to track adding tests, would prefer changes to be in the codeline.

**Pull request merged**